gcs/src/components/settingsModal.jsx

```
@@ -389,7 +439,24 @@ function FFmpegBinarySetting({ settingName }) {
   )
 }
 
-function Setting({ settingName, df }) {
+function Setting({ settingName, df, initialValue }) {
+  const { getSetting, setSetting } = useSettings()
+  const [changed, setChanged] = useState(false)
+  const [changedFromDefault, setChangedFromDefault] = useState(false)
+
+  useEffect(() => {
+    if (initialValue !== undefined)
+      setChanged(
+        JSON.stringify(getSetting(settingName)) != JSON.stringify(initialValue),
+      )
+
+    setChangedFromDefault(
+      JSON.stringify(getSetting(settingName)) != JSON.stringify(df.default),
+    )
+  }, [getSetting(settingName)])
+
+  const resetToDefault = () => setSetting(settingName, df.default)
```

Can you only reset to default if the default is not an empty string or an empty list? If it's an empty string or empty list that means there isn't a value set yet; so resetting to default is clearing the value - which isn't ideal as that's misleading